checkov/common/parallelizer/parallel_runner.py

```diff
@@ -55,7 +55,11 @@ def run_function(
 
     def _run_function_multiprocess_fork(
         self, func: Callable[[Any], _T], items: List[Any], group_size: Optional[int]
-    ) -> Generator[_T, None, None]:
+    ) -> Generator[_T, None, Iterable[_T]]:
+        if multiprocessing.current_process().daemon:
+            # can't fork, when already inside a pool
+            return self._run_function_multithreaded(func, items)  # noqa: B901
```

Comment:
@gruebel Mind explaining this piece here? When could it happen?

Reply:
when `spawn` mode is used and one of the childs tries to also run something in parallel, but with fork mode. I only saw this behaviour in the UTs, then sometimes it got messed up between `spawn` and `fork` mode.